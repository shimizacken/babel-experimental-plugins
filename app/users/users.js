export const getUserZip = (user) => user?.address?.zip;

export const isUserRent = (user) => user?.address?.isRent ?? 'not rent';