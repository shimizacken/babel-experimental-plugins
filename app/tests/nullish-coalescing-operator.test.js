import { isUserRent } from "../users/users";

describe('nullish coalescing operator test suite', () => {
  
    const users = [{
        name: "Jimi",
        address: {
            zip: 1628546,
            isHouse: false
        }
    }];
  
    it('should return undefined when nested object is undefined', () => {
      expect(isUserRent(users[1])).toMatchSnapshot();
    });

    it('should return zip when nested object is not undefined', () => {
        expect(isUserRent(users[0])).toMatchSnapshot();
      });
});
