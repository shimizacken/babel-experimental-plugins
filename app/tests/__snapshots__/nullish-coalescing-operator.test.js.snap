// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nullish coalescing operator test suite should return undefined when nested object is undefined 1`] = `"not rent"`;

exports[`nullish coalescing operator test suite should return zip when nested object is not undefined 1`] = `"not rent"`;
